src/test/java/guitests/PersonEditGuiTest.java

```diff
@@ -139,22 +135,57 @@ public void cancelOperation_usingAccelerator() {
-        assertEquals(pandaWongCardHandle, pandaWong);
+        assertEquals(pandaWongCardHandle, pandaWong); //Ensure correct state before cancelling.
+
         personListPanel.use_PERSON_CHANGE_CANCEL_ACCELERATOR();
         assertNull(personListPanel.getPersonCardHandle(pandaWong));
+    }
+
+    @Test
+    public void editPerson_editDuringGracePeriod() {
+        Person newAlice = new PersonBuilder(td.alice.copy()).withFirstName("Alicia").withLastName("Brownstone")
+                .withStreet("Updated street").withCity("Singapore").withPostalCode("123123")
+                .withBirthday("01.01.1979").withGithubUsername("alicebrown123").withTags(td.colleagues, td.friends).build();
+
+        //Get a reference to the card displaying Alice's details
+        PersonCardHandle alicePersonCard = personListPanel.selectCard(td.alice);
+
+        //Edit Alice to change to new values
+        EditPersonDialogHandle editPersonDialog = personListPanel.use_PERSON_EDIT_ACCELERATOR();
+        editPersonDialog.enterNewValues(newAlice).pressEnter();
 
-        //Ensure cancel operation has no side effects after grace period.
+        //Edit Alice again during pending state.
+        personListPanel.clickOnPerson(newAlice);
+        Person newerAlice = new PersonBuilder(newAlice.copy()).withFirstName("Felicia").withLastName("Yellowstone")
+                .withStreet("street updated").withCity("Malaysia").withPostalCode("321321")
+                .withBirthday("11.11.1979").withGithubUsername("yellowstone").withTags(td.colleagues).build();
+        editPersonDialog = personListPanel.use_PERSON_EDIT_ACCELERATOR();
+        editPersonDialog.enterNewValues(newerAlice).pressEnter();
+
+        //Ensure card is displaying Felicia before and after grace period.
+        assertEquals(alicePersonCard, newerAlice);
         sleepForGracePeriod();
-        assertNull(personListPanel.getPersonCardHandle(pandaWong));
-        personListPanel.isListMatching(td.alice, td.benson, td.charlie, td.dan, td.elizabeth);
+        assertEquals(alicePersonCard, newerAlice);
     }
 
-    //TODO: testing edits during grace period
+    @Test
+    public void cancelOperation_afterGracePeriod() {
+        //New
+        EditPersonDialogHandle addPersonDialog = personListPanel.clickNew();
+        Person pandaWong = new PersonBuilder("Panda", "Wong")
+                .withStreet("Chengdu Panda Street").withCity("Chengdu").withPostalCode("PANDA")
+                .withBirthday("01.01.1979").withGithubUsername("panda").withTags(td.colleagues, td.friends).build();
+        addPersonDialog.enterNewValues(pandaWong).clickOk();
+        sleepForGracePeriod();
+        personListPanel.clickOnListView();
+        personListPanel.use_LIST_GOTO_BOTTOM_SEQUENCE();
+        personListPanel.use_PERSON_CHANGE_CANCEL_ACCELERATOR();
+        assertEquals(personListPanel.getPersonCardHandle(pandaWong), pandaWong);
+    }
```

Comment:
To make testing efficient, you can test this scenario in another test that already has a completed edit op.
Inside this method simply have a comment explaining in which test this functionality is tested.